src/styles/index.scss

```diff
@@ -1,3 +1,90 @@
-body {
+* {
+  padding: 0;
   margin: 0;
+  box-sizing: border-box;
+}
+
+$size: 100px;
+$border: 1px solid black;
+$background: #eee;
+$text-color: #000;
+$text-size: 30px;
+$gap: 1px;
+$hover-color: #ffbfcb;
+$transition-duration: 0.5s;
+$width-calendar: 7 * $size + $gap * 6;
+$days: (
+  sun: 6,
+  mon: 0,
+  tue: 1,
+  wed: 2,
+  thu: 3,
+  fri: 4,
+  sat: 5,
+);
+
+.block_calendar {
+  width: 100%;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  font-family: Arial, Helvetica, sans-serif;
+  display: flex;
+  flex-wrap: wrap;
+  width: $width-calendar;
+  gap: $gap;
+
+  @each $day, $index in $days {
+    &--start-day-#{$day} {
+      .calendar__day:first-child {
+        margin-left: ($index * ($size + $gap));
+      }
+    }
```

Comment:
It's a good practice to avoid nesting when it's not necessary, as it can lead to more specific CSS selectors than needed, which can make it harder to override styles later. In this case, the nested `.calendar__day:first-child` could be written as `.calendar--start-day-#{$day} .calendar__day:first-child` without nesting.